engine/src/main/java/com/datatorrent/stram/engine/Reservoir.java

```diff
@@ -23,20 +23,26 @@
  *
  * @since 0.3.2
  */
-public interface Reservoir
+public interface Reservoir<T>
 {
   /**
    * the count of elements in this SweepableReservoir.
    *
    * @return the count
    */
-  public int size();
+  int size(final boolean dataTupleAware);
```

Comment:
@vrozov dataTupleAware: This Boolean tells that give the count of only data tuples and not control tuples.